test/tests/logger-prod.test.js

```diff
@@ -0,0 +1,264 @@
+      expect(_log.mock.calls.length).toEqual(0)
+    })
+
+    test('Log should contain HTTP method', async () => {
+      const query = gql`
+        {
+          AdminService {
+            Books {
+              nodes {
+                title
+              }
+            }
+          }
+        }
+      `
+      await GET(`/graphql?query=${query}`)
+      expect(_format(_log.mock.calls[0])).toContain('GET')
+    })
+
+    test('Log should not contain operationName if none was provided', async () => {
+      const query = gql`
+        {
+          AdminService {
+            Books {
+              nodes {
+                title
+              }
+            }
+          }
+        }
+      `
+      await GET(`/graphql?query=${query}`)
+      expect(_format(_log.mock.calls[0])).not.toContain('operationName')
+    })
+
+    test('Log should not contain variables if none were provided', async () => {
+      const query = gql`
+        {
+          AdminService {
+            Books {
+              nodes {
+                title
+              }
+            }
+          }
+        }
+      `
+      await GET(`/graphql?query=${query}`)
+      expect(_format(_log.mock.calls[0])).not.toContain('variables')
+    })
+
+    test('Log should contain operationName and its value', async () => {
+      const operationName = 'ListBooks'
+      const query = gql`
+        query ListAuthors {
+          AdminService {
+            Authors {
+              nodes {
+                name
+              }
+            }
+          }
+        }
+        query ${operationName} {
+          AdminService {
+            Books {
+              nodes {
+                title
+              }
+            }
+          }
+        }
+      `
+      await GET(`/graphql?operationName=${operationName}&query=${query}`)
+      expect(_format(_log.mock.calls[0])).toContain(`operationName: '${operationName}'`)
+    })
+
+    test('Log should not contain literal values', async () => {
+      const secretTitle = 'secret'
+      const query = gql`
+        query {
+          AdminService {
+            Books(filter: { title: { ne: "${secretTitle}"}}) {
+              nodes {
+                title
+              }
+            }
+          }
+        }
+      `
+      await GET(`/graphql?query=${query}`)
+      expect(_format(_log.mock.calls[0])).not.toContain(secretTitle)
+    })
+
+    test('Log should not contain variables or their values', async () => {
+      const secretTitle = 'secret'
+      const query = gql`
+        query ($filter: [AdminService_Books_filter]) {
+          AdminService {
+            Books(filter: $filter) {
+              nodes {
+                title
+              }
+            }
+          }
+        }
+      `
+      const variables = { filter: { title: { ne: secretTitle } } }
+      await GET(`/graphql?query=${query}&variables=${JSON.stringify(variables)}`)
+      expect(_format(_log.mock.calls[0])).not.toContain('$filter')
```

Comment:
`$filter` or `filter`?

Reply:
`$filter` is correct here, as in GraphQL all variable names need to be prefixed with `$`.

https://github.com/cap-js/graphql/blob/1d66c1d541c9c3463eb53c4ba0ae82d6f573334f/test/tests/logger-prod.test.js#L248-L250

This refers to the variable `$filter` that is defined in line 248 with type `[AdminService_Books_filter]` and used in line 250, where its value is passed to the argument `filter`. Any name could have been chosen by the request sender, so this could have also looked like:

```graphql
query ($foo: [AdminService_Books_filter]) {
  AdminService {
    Books(filter: $foo) {
      nodes {
        title
      }
    }
  }
}
```

with:

```javascript
const variables = { foo: { title: { ne: secretTitle } } }
```

The test ensures that the variable and its values don't appear in the log. If the argument itself (`filter`) would appear in the log, that wouldn't be a problem since it is known/defined in the GraphQL schema anyway.